src/Radicle/Daemon/HttpApi.hs

```diff
@@ -26,57 +33,90 @@ import qualified Paths_radicle as Radicle
 import           Radicle
 
 instance FromHttpApiData MachineId where
-  parseUrlPiece = Right . MachineId . toS
-  parseQueryParam = parseUrlPiece
+    parseUrlPiece = Right . MachineId . toS
+    parseQueryParam = parseUrlPiece
 
-singleKey :: Prelude.String -> (a -> b) -> (A.Object -> A.Parser a) -> A.Value -> A.Parser b
-singleKey tyName c k = A.withObject tyName $ k >=> pure . c
+instance ToHttpApiData MachineId where
+    toUrlPiece (MachineId id) = id
 
 newtype QueryRequest = QueryRequest { expression :: Value }
   deriving (Generic)
 
 instance A.ToJSON QueryRequest where
-  toJSON = valueToJson . expression
+    toJSON QueryRequest{..} = A.object
```

Comment:
Can we add a [validateEveryToJSON](http://hackage.haskell.org/package/servant-swagger-1.1.7/docs/Servant-Swagger-Test.html#v:validateEveryToJSON) test so we're sure the swagger instances are correct?

Reply:
I created a ticket for that.